caches/clan.go

```diff
@@ -0,0 +1,83 @@
+package caches
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/topfreegames/khan/models"
+)
+
+// ClansSummaries represents a cache for the RetrieveClansSummaries operation.
+type ClansSummaries struct {
+	// Cache points to an instance of gocache.Cache used as the backend cache object.
+	Cache *gocache.Cache
+
+	// TTL is used to compute the amount of time that a clan payload will live in the cache: a random integer in [TTL, 2 * TTL).
+	TTL time.Duration
+}
+
+// GetClansSummaries is a cache in front of models.GetClansSummaries() with the exact same interface.
+// Like models.GetClansSummaries(), this function may return partial results + CouldNotFindAllClansError.
+// The map[string]interface{} return type represents a summary of one clan with the following keys/values:
+// "membershipCount":  int
+// "publicID":         string
+// "metadata":         map[string]interface{} (user-defined arbitrary JSON object with clan metadata)
+// "name":             string
+// "allowApplication": bool
+// "autoJoin":         bool
+// TODO: replace this map with a richer type
```

Comment:
FYI, I usually suggest using `TODO($USER): $message`, where $USER is the person who has more context about the TODO and what it means (not necessarily the person expected to fix it - anyone could do that, ideally).

Reply:
Done!